code/game/machinery/exonet_node.dm

```diff
@@ -0,0 +1,146 @@
+/obj/machinery/exonet_node
+	name = "exonet node"
+	desc = null // Gets written in New()
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "exonet_node"
+	idle_power_usage = 25
+	var/on = 1
+	var/toggle = 1
+	density = TRUE
+	anchored = TRUE
+	circuit = /obj/item/circuitboard/machine/exonet_node
+	max_integrity = 300
+	integrity_failure = 100
+	armor = list(melee = 20, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 50, acid = 70)
+	var/opened = 0
+
+	var/list/logs = list() // Gets written to by exonet's send_message() function.
+
+// Proc: New()
+// Parameters: None
+// Description: Adds components to the machine for deconstruction.
+/obj/machinery/exonet_node/map/New()
+	..()
+
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/machine/exonet_node(src)
+	component_parts += new /obj/item/stock_parts/subspace/ansible(src)
```

Comment:
A seriously outdated way to add machine parts. As far as I remember, you can just set circuit board typepath, and machine code will handle spawning the components by itself.